src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java

```diff
@@ -151,7 +151,39 @@ protected final void init() {
         for (Trigger t : triggers) {
             t.start(this, Items.currentlyUpdatingByXml());
         }
-        loadComputation();
+        synchronized (this) {
+            computation = createComputation(null);
+        }
+    }
+
+    @Override
+    public void onCreatedFromScratch() {
+        try {
+            FileUtils.forceMkdir(getComputationDir());
+        } catch (IOException x) {
+            LOGGER.log(Level.WARNING, null, x);
+        }
+        super.onCreatedFromScratch();
+    }
+
+    @Override
+    public void onLoad(ItemGroup<? extends Item> parent, String name) throws IOException {
+        super.onLoad(parent, name);
+        try {
+            FileUtils.forceMkdir(getComputationDir());
+        } catch (IOException x) {
+            LOGGER.log(Level.WARNING, null, x);
+        }
+        synchronized (this) {
+            XmlFile file = computation.getDataFile();
+            if (file.exists()) {
+                try {
+                    file.unmarshal(computation);
```

Comment:
Same root. Without moving this here `OrganizationFolder` has issues with it's `MultibranchProject` children (didn't affect when we were actually changing the name to the mangled name, but with this change we enable the disconnect and that surfaces the issue)